### PR TITLE
treewide: fix AppendToName

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#68519

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by lukegb — APPROVED** [review]
Mercurial seems fine.

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 161414` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>gawkInteractive</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>clevis</li>
    <li>davix-copy</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>gnudatalanguage</li>
    <li>highfive-mpi</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>luksmeta</li>
    <li>mercurialFull</li>
    <li>netcdf-mpi</li>
    <li>neuron-full</li>
    <li>neuron-mpi</li>
    <li>python310Packages.neuron-mpi</li>
    <li>python39Packages.neuron-mpi</li>
    <li>raxml-mpi</li>
    <li>sqlite-interactive</li>
    <li>tang</li>
    <li>udiskie</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20115,7 +20110,7 @@ with pkgs;
 
   sqlitecpp = callPackage ../development/libraries/sqlitecpp { };
 
-  sqlite-interactive = appendToName "interactive" (sqlite.override { interactive = true; }).bin;
+  sqlite-interactive = sqlite.override { interactive = true; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:20113]: Do we need to re-add the .bin?

Comment (Stunkymonkey) [reply pkgs/top-level/all-packages.nix]: good catch. should be fixed now

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**